source/org/zfin/mutant/repository/HibernateMutantRepository.java

```diff
@@ -170,203 +165,206 @@ public Object transformTuple(Object[] tuple, String[] aliases) {
+                            }
+                        }
+                    }
+                    hql += "ORDER BY fishox.fish.order, fishox.fish.nameOrder ";
+
+                    Query query = session.createQuery(hql);
+                    query.setParameter("aoTerm", item);
+                    query.setParameter("tag", PhenotypeStatement.Tag.NORMAL.toString());
+                    if (MapUtils.isNotEmpty(bean.getFilterMap())) {
+                        for (Map.Entry<String, String> entry : bean.getFilterMap().entrySet()) {
+                            query.setParameter(entry.getKey(), "%" + entry.getValue().toLowerCase() + "%");
+                        }
+                    }
+                    // have to add extra select because of ordering, but we only want to return the first
+                    setTupleResultTransformer(query, (Object[] tuple, String[] aliases) -> tuple[0]);
+                    return PaginationResultFactory.createResultFromScrollableResultAndClose(bean, query.scroll());
                 }
-            }
-        }
-        hql += "ORDER BY fishox.fish.order, fishox.fish.nameOrder ";
 
-        Query query = session.createQuery(hql);
-        query.setParameter("aoTerm", item);
-        query.setParameter("tag", PhenotypeStatement.Tag.NORMAL.toString());
-        if (MapUtils.isNotEmpty(bean.getFilterMap())) {
-            for (Map.Entry<String, String> entry : bean.getFilterMap().entrySet()) {
-                query.setParameter(entry.getKey(), "%" + entry.getValue().toLowerCase() + "%");
-            }
-        }
-        // have to add extra select because of ordering, but we only want to return the first
-        query.setResultTransformer(new BasicTransformerAdapter() {
-            @Override
-            public Object transformTuple(Object[] tuple, String[] aliases) {
-                return tuple[0];
-            }
-        });
-        return PaginationResultFactory.createResultFromScrollableResultAndClose(bean, query.scroll());
-    }
+                @SuppressWarnings({"unchecked"})
+                public List<Genotype> getGenotypesByFeature(Feature feature) {
+                    Session session = currentSession();
 
-    @SuppressWarnings({"unchecked"})
-    public List<Genotype> getGenotypesByFeature(Feature feature) {
-        Session session = HibernateUtil.currentSession();
+                    String hql = """ 
+                        select distinct geno from Genotype geno, GenotypeFeature genofeat 
+                        WHERE genofeat.feature = :feature 
+                        AND genofeat.genotype = geno 
+                        ORDER BY geno.nameOrder 
+                        """;
 
-        String hql = "select  distinct geno from Genotype geno, GenotypeFeature genofeat " +
-            "WHERE  genofeat.feature = :feature " +
-            "AND genofeat.genotype =geno " +
-            "ORDER BY geno.nameOrder";
+                    Query query = session.createQuery(hql);
```

Comment:
Retrieve generic Query
query<Genotype> query = session.createQuery(hql, Genotype.class)

Reply:
Thanks for the comment.  I added types to each of those queries.